src/Symfony/Component/Form/Tests/Extension/Core/Type/CheckboxTypeTest.php

```diff
@@ -170,4 +170,31 @@ public function provideCustomModelTransformerData()
             array('unchecked', false),
         );
     }
+
+    public function testSubmitNullWithEmptyDataTrue()
+    {
+        $form = $this->factory->create('checkbox', null, array(
+            'empty_data' => true,
```

Comment:
Actually, this code is invalid. `empty_data` expects data in view format, not in model format. The view format depends on the transformer that is used.

Imagine the field is configured with the following transformer

~~~php
true <---> 'yes'
false <---> 'no'
~~~

Then this code will fail. The `empty_data` value `true` will be cast to the string `'1'`. Consequently, the transformer fails, since it doesn't know how to reverse transform `'1'`.

The correct test would be:

~~~php
$form = $this->factory->create('checkbox', null, array(
    'empty_data' => '1',
));
~~~

I understand this is not very straight-forward nor intuitive, but we can't change that for the moment.

I believe that the use case that motivated this change was actually using the `empty_data` incorrectly, by passing data in model instead of view format (ping @guilhermeblanco).